das/metrics.go

```diff
@@ -169,6 +172,9 @@ func (m *metrics) observeGetHeader(ctx context.Context, d time.Duration) {
 	if m == nil {
 		return
 	}
+	if ctx.Err() != nil {
+		ctx = context.Background()
+	}
```

Comment:
@walldiss, I mean for this case. The operation was canceled, and you still wanted to track it, but we should not track it as a successful one as we do right now or not track it at all due to your argument:
> And there is not much to worry about anyway, since metrics are not meant to be precise, it is statistical tool and it should not be a problem if we lose few.

Reply:
it is canceled for original operation, not for metric observation. Why would we respect original cancel for observing canceled call?

Reply:
> Why would we respect original cancel for observing canceled call?

I am not saying that we need to __respect__ the original cancel. The code here submits metrics for both canceled and normal operations. My point is to separate canceled operations from successful ones.